src/main/java/org/opensrp/service/formSubmission/EventsListener.java

```diff
@@ -93,7 +95,7 @@ public void processEvent() {
 			
 			for (Event event : events) {
 				try {
-					event = eventService.processOutOfArea(event);
+					event = eventService.processOutOfArea(event, username);
```

Comment:
You can pass `event.getProvider` as username here

Reply:
Passing event.getProviderId()